src/sorta.nim

```diff
@@ -523,6 +573,38 @@ when isMainModule:
     assert st.getOrDefault("www.dell.com") == "143.166.224.230"
     assert(st.entries == 16)
 
+    var keys = toSeq(st.keys())
+    for i in 0..<keys.len:
+      var key = "" & keys[i]
+      proc collect(): seq[string] = 
+        result = @[]
+        for position in st.scanFrom(key):
+          result.add(position.key)
+      echo "Checking key " & key
+      assert collect() == keys[i..<keys.len]
+      key[key.len-1].dec
+      echo "Checking key " & key
+      assert collect() == keys[i..<keys.len]
+      key = "" & keys[i]
+      key[key.len-1].inc
+      echo "Checking key " & key
+      assert collect() == keys[i+1..<keys.len]
+
+    for i in 0..<keys.len:
+      var from_key = "" & keys[i]
```

Comment:
Why is this concatenation needed?

(In other places above and below too)

Reply:
I am going to overwrite the last character to test non exact keys.
Is there a more idiomatic way to make a copy of a string?

Reply:
> Is there a more idiomatic way to make a copy of a string?

Have you tried if a simple `var from_key = keys[i]` works as intended?

Reply:
I just tested it and that in fact does work. I guessed that strings were references objects and that I now see is wrong :-)  In any case, I refactored the test code to explicitly generate new strings by appending a value or chopping off the last value because that seemed more clear. Thanks!